modules/dnn/src/layers/slice_layer.cpp

```diff
@@ -678,12 +747,21 @@ class SliceLayerImpl : public SliceLayer
         }
     }
 
+    template <typename T>
+    void flip(Mat& output) // break if 1d tensor?
+    {
+        for (int i = 0; i < negative_steps_flag.size(); ++i)
+            if (negative_steps_flag[i] != 0)
+                cv::flipND(output, output, i);
+    }
 protected:
     // The actual non-negative values determined from @p sliceRanges depends on input size.
     std::vector<std::vector<Range> > finalSliceRanges;
+    std::vector<int> negative_steps_flag;
```

Comment:
We can save only negative dims here. Or use smaller type - `uint8_t` or even `bool`(`bool` needs testing).

Reply:
Done with saving only dims that have neg steps.